.github/workflows/build-and-test.yaml

```diff
@@ -31,15 +31,15 @@ jobs:
     name: Build sdist wheel
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - run: |
           make preprocess
           pipx run build --sdist
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
```

Comment:
Just curious: why are you making all these unrelated changes in this PR? you know v4 of `actions/upload-artifact` has [breaking changes](https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md), right?

Reply:
Thanks for pointing out the changes to v4 actions/upload-artifact. I've been experimenting with different versions and configurations to see why GitHub Actions was causing errors. The commits you saw were part of those experiments. I really appreciate your feedback and hope you and also continue to help with development.